lib/auth/auth.js

```diff
@@ -44,20 +46,44 @@ export const getSessionFromCookies = async () => {
   if (!idToken) return DEFAULT_SESSION
 
   const user = jwtDecode(idToken)
-  const isAuthenticated = user && user.email_verified
 
+  // Check if token is not expired on client side.
+  // This may happen if the users token expires while the are navigating between pages that
+  // are not server side rendered.
+  // NOTE: This does not check if the token has been signed by the correct signing key,
+  // we do not need to verify the JWT on the client side as if an attacker were forging a signed JWT
+  // the server will reject all API and page load requests.
+  const jwtValid = isJwtExpValid(user.exp)
+  if (!jwtValid) {
+    const redirectUrl = encodeURIComponent(getPathAndHash())
+    window.location.replace(`${getBaseUrl()}/auth/sign-thru?redirect=${redirectUrl}`)
```

Comment:
This is simpler using the Router 
import Router from 'next/router'
    const redirectUrl = encodeURIComponent(getPathAndHash())
    const signThruUrl = `/auth/sign-thru?redirect=${redirectUrl}`
    Router.push(signThruUrl)